src/chunk.rs

```diff
@@ -68,6 +68,57 @@ impl StrChunk {
         self.bytes.is_empty()
     }
 
+    /// Views UTF-8 content from a byte buffer.
+    ///
+    /// Builds a `StrChunk` from the longest leading part of the `Bytes` view that
+    /// validates as UTF-8. The new `StrChunk` contains a reference to the slice that is valid UTF-8.
+    ///
+    /// # Errors
+    ///
+    /// If an invalid UTF-8 sequence is encountered within `src`, an error
+    /// value with recovery information is returned in the `Err` variant.
+    /// The valid UTF-8 part preceding the invalid sequence is taken
+    /// out of `src` and can be obtained from the `ExtractUtf8Error` value.
+    ///
+    /// # Example
+    ///
+    /// Basic usage:
+    ///
+    /// ```rust
+    /// # use bytes::Bytes;
+    /// # use strchunk::StrChunk;
+    /// let s1: &[u8] = b"\xd0\x97\xd0\xb4\xd1\x80\xd0\xb0\xd0";
+    /// let mut buf = Bytes::from(s1);
+    ///
+    /// let chunk = StrChunk::view_utf8(buf.clone()).unwrap();
```

Comment:
What's the use case here? Should the caller be able to access the trailing bytes without extra operations? Something like:

```rust
let (s, tail) = StrChunk::split_utf8(buf).unwrap();
```